store/repository.go

```diff
@@ -565,17 +568,25 @@ func patchRepositoryImpl(ctx context.Context, tx *Tx, patch *api.RepositoryPatch
 	if v := patch.RefreshToken; v != nil {
 		set, args = append(set, fmt.Sprintf("refresh_token = $%d", len(args)+1)), append(args, *v)
 	}
-
-	args = append(args, patch.ID)
+	where := []string{}
+	if v := patch.ID; v != nil {
+		where, args = append(where, fmt.Sprintf("id = $%d", len(args)+1)), append(args, *v)
+	}
+	if v := patch.WebURL; v != nil {
+		where, args = append(where, fmt.Sprintf("web_url = $%d", len(args)+1)), append(args, *v)
+	}
+	if len(where) == 0 {
+		return nil, common.Errorf(common.DbMissingWhereClause, "missing where clause for patching repository")
```

Comment:
And for this one, I would prefer common.Invalid

Specific error code like DbMissingWhereClause is used if it's a user generated error. But here, this is a code logic error, so no need to create a special error code.

Reply:
Improve it in #2686. PTAL there.